src/main/java/com/simibubi/create/content/contraptions/fluids/actors/FillingRecipe.java

```diff
@@ -68,13 +74,27 @@ public void addAssemblyFluidIngredients(List<FluidIngredient> list) {
 	@Override
 	@Environment(EnvType.CLIENT)
 	public Component getDescriptionForAssembly() {
-		List<FluidStack> matchingFluidStacks = fluidIngredients.get(0)
-			.getMatchingFluidStacks();
+		List<FluidStack> matchingFluidStacks = fluidIngredients.get(0).getMatchingFluidStacks();
+
 		if (matchingFluidStacks.size() == 0)
 			return new TextComponent("Invalid");
-		return Lang.translate("recipe.assembly.spout_filling_fluid",
-			new TranslatableComponent(matchingFluidStacks.get(0)
-				.getTranslationKey()).getString());
+
+		Fluid fluid = matchingFluidStacks.get(0).getFluid();
+		String translationKey;
+		if (fluid == Fluids.EMPTY) {
+			translationKey = "";
+		} else if (fluid == Fluids.WATER) {
+			translationKey = "block.minecraft.water";
+		} else if (fluid == Fluids.LAVA) {
+			translationKey = "block.minecraft.lava";
+		} else {
+			ResourceLocation id = Registry.FLUID.getKey(fluid);
+			String key = Util.makeDescriptionId("block", id);
+			String translated = I18n.get(key);
+			translationKey = translated.equals(key) ? Util.makeDescriptionId("fluid", id) : key;
+		}
+
+		return Lang.translate("recipe.assembly.spout_filling_fluid", new TranslatableComponent(translationKey).getString());
```

Comment:
I'd like this code de-duplicated and moved into a util class, probably FluidUtil

Reply:
yeah, I agree with you, just didnt knew where to put it, so a general, generic, approach was using duplicated code